eclair-core/src/main/scala/fr/acinq/eclair/package.scala

```diff
@@ -30,11 +31,24 @@ package object eclair {
    * We are using 'new SecureRandom()' instead of 'SecureRandom.getInstanceStrong()' because the latter can hang on Linux
    * See http://bugs.java.com/view_bug.do?bug_id=6521844 and https://tersesystems.com/2015/12/17/the-right-way-to-use-securerandom/
    */
-  val secureRandom = new SecureRandom()
+  private val secureRandom = new SecureRandom()
+
+  def initRandom(): Unit = secureRandom.nextInt()
+
+  /**
+   * We're using an additional, weaker randomness source to protect against catastrophic failures of the SecureRandom
+   * instance.
+   */
+  val weakRandom = new WeakRandom()
 
   def randomBytes(length: Int): ByteVector = {
     val buffer = new Array[Byte](length)
     secureRandom.nextBytes(buffer)
+    val weakBuffer = new Array[Byte](length)
+    weakRandom.nextBytes(weakBuffer)
+    for (i <- buffer.indices) {
+      buffer(i) = (buffer(i) ^ weakBuffer(i)).toByte
+    }
     ByteVector.view(buffer)
```

Comment:
```suggestion
    val rand1 = {
      val buffer = new Array[Byte](length)
      secureRandom.nextBytes(buffer)
      ByteVector.view(buffer)
    }
    val rand2 = {
      val weakBuffer = new Array[Byte](length)
      weakRandom.nextBytes(weakBuffer)
      ByteVector.view(weakBuffer)
    }
    rand1 xor rand2
```

Reply:
This requires two byte array allocation for no reason, we can reuse at least one of them.
And xoring two `ByteVector` has to be less efficient than just iterating over the array to xor each byte individually.

Reply:
TBH I think this is really clear enough with the current code, and there's no risk of mutating the `ByteVector`, it's only applied after xor-ing everything.